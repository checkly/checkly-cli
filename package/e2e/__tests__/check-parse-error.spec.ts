import { runChecklyCli } from '../run-checkly'

describe('check parse error', () => {
  it('"checkly test" should return a clear error when there are check dependency errors', () => {
    const result = runChecklyCli({
      args: ['test'],
      directory: 'check-parse-error'
    })
    expect(result.stderr).toMatchSnapshot()
  })
})
