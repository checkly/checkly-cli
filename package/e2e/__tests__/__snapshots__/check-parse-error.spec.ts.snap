// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check parse error "checkly test" should return a clear error when there are check dependency errors 1`] = `
"    Error: Encountered an error parsing check files for entrypoint.js.

    The following dependencies weren't found:
    	nonexistent-file.js
    	nonexistent-file-2.js


    The following NPM dependencies were used, but aren't supported in the 
    runtimes.
    For more information, see https://www.checklyhq.com/docs/runtimes/.
    	entrypoint.js imports unsupported dependencies:
    		unsupported-dependency
    		unsupported-dependency-2


    The following files couldn't be parsed:
    	syntax-error-1.js - Unexpected token (3:0)
    	syntax-error-2.js - Unexpected token (2:0)

"
`;
